lib/spack/spack/util/path.py

```diff
@@ -68,11 +68,12 @@ def repl(match):
     return re.sub(r'(\$\w+\b|\$\{\w+\})', repl, path)
 
 
-def canonicalize_path(path):
+def canonicalize_path(path, make_abs=True):
```

Comment:
instead of adding an optional argument, factor out `substitute_variables()` (which does everything but the absolute path) and make `canonicalize_path()` call that.